### PR TITLE
If both keepSynced and notifyUpdate are true, only deliver ops while …

### PR DESCRIPTION
**Pull request by galganif**
…the proxy is synchronized

### CODE REVIEW

File: src/runtime/storageNG/handle.ts [hunk 1]
```diff
@@ -123,6 +123,7 @@ export abstract class Handle<StorageType extends CRDTTypeRecord> {
   // - notifySync (bool): if keepSynced is true, call onHandleSync when the full data is received
   // - notifyUpdate (bool): call onHandleUpdate for every change event received
   // - notifyDesync (bool): if keepSynced is true, call onHandleDesync when desync is detected
+  // If both keepSynced and notifyUpdate are true, updates are only delivered while the proxy is synchronized.
```
Comment (mykmartin) [inline src/runtime/storageNG/handle.ts]: I'm not sure we actually want this behaviour. Can you give some details on the motivation for this?

Comment (mykmartin) [reply src/runtime/storageNG/handle.ts]: As per offline conversation, this *is* probably the right thing, but I would suggest updating the comment inline:
```
// - notifyUpdate (bool): if keepSynced is true, call onHandleUpdate for every change event received if the proxy is currently synchronized (i.e. updates received when desynced are ignored); if keepSynced is false, just notify for all updates
```

File: src/runtime/storageNG/storage-proxy.ts [hunk 2]
```diff
@@ -198,7 +199,8 @@ export class StorageProxy<T extends CRDTTypeRecord> {
   protected notifyUpdate(operation: CRDTOperation) {
     const version: VersionMap = this.versionCopy();
     for (const handle of this.handles) {
-      if (handle.options.notifyUpdate) {
+      if (handle.options.notifyUpdate &&
+          (handle.options.keepSynced === false || this.synchronized)) {
```
Comment (mykmartin) [inline src/runtime/storageNG/storage-proxy.ts]: `!handle.options.keepSynced`

File: src/runtime/storageNG/storage-proxy.ts [hunk 3]
```diff
@@ -174,6 +174,7 @@ export class StorageProxy<T extends CRDTTypeRecord> {
           if (!this.crdt.applyOperation(op)) {
             // If we cannot cleanly apply ops, sync the whole model.
             this.clearSynchronized();
+            this.notifyUpdate(op);
```
Comment (mykmartin) [inline src/runtime/storageNG/storage-proxy.ts]: This case should do a first pass over all handles to notify those with `!keepSynced && notifyUpdate`, then there's no need to have this new `notifyUpdate` call.

Comment (galganif) [reply src/runtime/storageNG/storage-proxy.ts]: done

**Pull request merged**